crates/cheatcodes/defs/src/vm.rs

```diff
@@ -134,6 +144,49 @@ interface Vm {
         bytes stderr;
     }
 
+    /// The result of a `getStateDiff` call.
+    struct AccountAccess {
+        /// What accessed the account.
+        address accessor;
+        /// The account whose storage was accessed.
+        /// It's either the account created, callee or a selfdestruct recipient for CREATE, CALL or SELFDESTRUCT.
+        address account;
+        /// The kind of account access.
+        AccountAccessKind kind;
+        /// If the account is initialized or empty
+        bool initialized;
+        /// The previous account balance of the accessed account.
+        uint256 oldBalance;
+        /// The new account balance of the accessed account. This represents the balance change ignoring any reverts that may have occurred.
+        uint256 newBalance;
```

Comment:
> This represents the balance change ignoring any reverts that may have occurred.

So this is really a *potential* new balance that is only accurate if the call did not revert, and not accurate if the call reverted?

Reply:
Correct. So if account A calls `B.foo{value: 1 ether}()` and then A later reverts, its `newBalance` will be 1 eth short.

Reply:
Clarified this such that it uses "potential" and reads cleaner.